### PR TITLE
Adds -iree-tf-strip-*-metadata passes that remove tf attributes.

### LINKED ISSUE #3965: Strip all the tensorflow junk attributes from the IR prior to bringing into IREE
It's really hard to read IR that originates from tensorflow due to all the attributes that get carried along. We don't need any of them once the signatures have been generated (and may not even need them for that). We should have a pass on the tensorflow integrations side that strips them all prior to lowering into IREE's core input dialects like flow.

### PR DESCRIPTION
**Pull request by benvanik**
me

While here I noticed that this directory is a mismatch of styles from
MLIR and TF, wasn't properly namespaced (both on pass names or C++) in
many places, and didn't follow the rest of the conventions used in
iree/compiler/. To make this code easier to maintain and before we start
to copypasta it for other integrations I did a first-pass cleanup.
I also noticed we still had a bunch of super flakey alwayslink stuff so
I fixed that too. No more alwayslink required!

Fixes #3965.

### CODE REVIEW

File: integrations/tensorflow/compiler/Passes.cpp [hunk 1]
```diff
@@ -25,56 +25,64 @@
...
+namespace TF {
+
+void registerAllDialects(mlir::DialectRegistry &registry) {
+  registry.insert<tf_tensorlist::TFTensorListDialect,
+                  tf_strings::TFStringsDialect>();
+}
 
 // All IREE-specific passes that lower TF representations before reaching the
 // IREE core should go here.
-void createIreeTfImportPipeline(OpPassManager &pm) {
```
Comment (ScottTodd) [inline integrations/tensorflow/compiler/Passes.cpp:31]: @stellaraccident this (and possibly other files) will conflict with https://github.com/google/iree/pull/4043

**Review by ScottTodd — APPROVED** [review]
Sorry for the delayed reviews. This generally LGTM. 

* Any thoughts, @rsuderman ?
* This has some conflicts with tip of tree and with @stellaraccident 's pending https://github.com/google/iree/pull/4043 . How do you all want to proceed?

File: integrations/tensorflow/compiler/Passes.cpp (hunk 1 above)
Comment (stellaraccident) [reply integrations/tensorflow/compiler/Passes.cpp:31]: FIFO - go ahead and land and I'll rebase and likely submit on the OSS side (thought there were going to be more internal entanglements than I thought so started there). I can wait ~a day.

Comment (ScottTodd) [reply integrations/tensorflow/compiler/Passes.cpp:31]: SGTM. I just resolved the merge conflicts with @rsuderman 's https://github.com/google/iree/pull/3941. Planning on merging once checks come back green.

Comment (stellaraccident) [reply integrations/tensorflow/compiler/Passes.cpp:31]: Thanks - I was going to offer to help land it, but it seems like you've already done the work :)

Comment (rsuderman) [reply integrations/tensorflow/compiler/Passes.cpp:31]: > Sorry for the delayed reviews. This generally LGTM.
> 
> * Any thoughts, @rsuderman ?
> * This has some conflicts with tip of tree and with @stellaraccident 's pending #4043 . How do you all want to proceed?

Looks fine with me assuming it does not conflict too much with Stella's changes. I initially hesitated at removing the metadata until I realized its long after I need it for shape inference.

Comment (stellaraccident) [reply integrations/tensorflow/compiler/Passes.cpp:31]: It conflicts but I'll rebase (and had already locally fixed up some of the style issues so hopefully not too bad) and send a fresh PR on the main branch when ready.

**Pull request merged**